File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int additionalRocks) {
        vector<int> rem;
        int len = rocks.size();
        for(int i = 0; i < len; i++) {
            rem.push_back(capacity[i] - rocks[i]);
        }
        sort(rem.begin(), rem.end());
        int bags = 0;
        for(int i = 0; i < len; i++) {
            if (rem[i] > additionalRocks) break;
            additionalRocks -= rem[i];
            bags++;
        }
        return bags;
    }
};

File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        rem = []
        for c, curr in zip(capacity, rocks):
            rem.append(c - curr)
        
        rem.sort()
        
        max_bags = 0
        for _, value in enumerate(rem):
            if value > additionalRocks:
                break
            max_bags += 1
            additionalRocks -= value
            
        return max_bags
            
        